x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -605,24 +605,42 @@ public void removeFromCache(KeyType cacheKey) {
         forceEvict(cacheKey::equals);
     }
 
-    public void forceEvict(Predicate<KeyType> cacheKeyPredicate) {
+    /**
+     * Evicts entries from the cache that match the given predicate.
+     *
+     * @param cacheKeyPredicate
+     * @return The number of entries evicted from the keyMapping.
+     */
+    public int forceEvict(Predicate<KeyType> cacheKeyPredicate) {
         final List<Entry<CacheFileRegion>> matchingEntries = new ArrayList<>();
         keyMapping.forEach((key, value) -> {
             if (cacheKeyPredicate.test(key.file)) {
                 matchingEntries.add(value);
             }
         });
+        var evictedCount = 0;
         if (matchingEntries.isEmpty() == false) {
             synchronized (this) {
                 for (Entry<CacheFileRegion> entry : matchingEntries) {
                     boolean evicted = entry.chunk.forceEvict();
                     if (evicted && entry.chunk.sharedBytesPos != -1) {
                         unlink(entry);
                         keyMapping.remove(entry.chunk.regionKey, entry);
+                        evictedCount++;
                     }
                 }
             }
         }
+        return evictedCount;
+    }
+
+    /**
+     * Evicts all entries from the cache.
+     *
+     * @return The number of entries evicted from the keyMapping.
+     */
+    public int forceEvictAll() {
```

Comment:
Would it be doable to add a unit test for this new method? And possibly update existing tests for forceEvict to verify the returned value?

Side note, for my curiosity: do we need a new public method? We could also easily provide an always true predicate?

Reply:
> do we need a new public method? We could also easily provide an always true predicate?

I've gone back and forth on this, see  https://github.com/elastic/elasticsearch-serverless/pull/462#discussion_r1259499244

Unfortunately, it seems the only location where `forceEvict()` is called from a test is here: https://github.com/JVerwolf/elasticsearch/blob/989b5cfb23309d75012b1f83ddc6368c67927405/x-pack/plugin/blob-cache/src/test/java/org/elasticsearch/blobcache/shared/SharedBlobCacheServiceTests.java#L278
https://github.com/elastic/elasticsearch/blob/989b5cfb23309d75012b1f83ddc6368c67927405/x-pack/plugin/blob-cache/src/test/java/org/elasticsearch/blobcache/shared/SharedBlobCacheServiceTests.java#L278 

I'm currently thinking of reverting this to use `forceEvict()`, as adding a new method to this cache requires doing the  due-diligence of exhaustively testing it, probably in multi-threaded scenarios. If so, this adds more overhead to this PR than I think is worth it given the nature of these changes. WDYT?

Reply:
I agree. if the existing tests already offer enough coverage, that seems like a pragmatic solution.